File: src/main/java/org/dxworks/githubminer/dto/export/PullRequestReviewDTO.java
package org.dxworks.githubminer.dto.export;

import com.google.api.client.util.Key;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.dxworks.githubminer.dto.response.repository.pullrequests.PullRequestReview;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PullRequestReviewDTO {
    @Key
    private UserDTO user;
    @Key
    private String state;
    @Key
    private String body;
    @Key
    private String date;

    public static PullRequestReviewDTO fromPullRequestReview(PullRequestReview pullRequestReview) {
        if (pullRequestReview == null || com.google.api.client.util.Data.isNull(pullRequestReview))
            return null;
        return builder()
                .body(pullRequestReview.getBody())
                .date(pullRequestReview.getSubmittedAt())
                .state(pullRequestReview.getState())
                .user(UserDTO.fromUser(pullRequestReview.getUser()))
                .build();
    }
}

File: src/main/kotlin/org/dxworks/githubminer/dto/export/PullRequestReviewDTO.kt
package org.dxworks.githubminer.dto.export

import com.google.api.client.util.Data
import com.google.api.client.util.Key
import org.dxworks.githubminer.dto.response.repository.pullrequests.PullRequestReview

class PullRequestReviewDTO {
    @Key
    var user: UserDTO? = null

    @Key
    var state: String? = null

    @Key
    var body: String? = null

    @Key
    var date: String? = null

    companion object {
        fun fromPullRequestReview(pullRequestReview: PullRequestReview?): PullRequestReviewDTO? {
            return if (pullRequestReview == null || Data.isNull(pullRequestReview)) null else PullRequestReviewDTO().apply {

                    body = pullRequestReview.body
                    date = pullRequestReview.submittedAt
                    state = pullRequestReview.state
                    user = UserDTO.Companion.fromUser(pullRequestReview.user)
            }
        }
    }
}
